File: Yard/Models/Deploy.cs
using System.Collections.Generic;

namespace Yard.Models
{
    public class Deploy
    {
        public string Environment { get; set; }
        public string Application { get; set; }
        public string Version { get; set; }
        public string BuildRevision { get; set; }
        public string PreviousBuildRevision { get; set; }
        public List<PullRequest> PullRequests { get; set; }
        public string Status { get; set; }
        public string ResultsUrl { get; set; }
    }
}

File: Yard/ClientApp/src/components/Deploy.ts
import {PullRequest, PullRequestJSON} from "./PullRequest";

export interface Deploy {
    environment: string,
    application: string,
    version: string,
    buildRevision: string,
    previousBuildRevision: string,
    pullRequests: PullRequest[],
    status: string,
    resultsUrl: string
}

export interface DeployJSON {
    Environment: string,
    Application: string,
    Version: string,
    BuildRevision: string,
    PreviousBuildRevision: string,
    PullRequests: PullRequestJSON[],
    Status: string,
    ResultsUrl: string
}